[Andre Lin HuiKai] iP

4ndrelim opened the pull request:
# MyDuke
> _"Building the damned plane as we fly it"_ - __lima_dingdong__ [[look up](https://github.com/4ndrelim)]

__MyDuke__ ensures you are on track in completing your tasks so your plane do not crash and burn! It is:
- ***COMPLETELY FREE***
- text-based
- easy to learn
- reliable
- __MOST IMPORTANTLY__, ~~FAST~~ _SUPER FAST_ to use

To use, simply,
1. [git clone](https://git-scm.com/book/en/v2/Git-Basics-Getting-a-Git-Repository) from [here](https://github.com/4ndrelim/ip)
2. Navigate to `build/libs`
3. run `java -jar Duke.jar` 
4. add your tasks
5. let it manage! 😄 

__Features__:
- [x] Managing tasks
- [x] Saving tasks
- [x] Searching tasks
- [ ] Ordering tasks (_Coming soon_)

If you are a Java programmer, feel free to examine and improve the current product! Here's the `main` method:
```
public class Main {
    public static void main(String[] args) {
        Duke duke = new Duke();
        duke.run();
    }
}
```

Commit:
Add Gradle support

4ndrelim committed:
Implement greet, echo, and exit functionalities

4ndrelim committed:
Add helper method to format user input

4ndrelim committed:
Create Task class and TaskList class to manage all tasks

4ndrelim committed:
Fix numbering when displaying list of tasks

4ndrelim committed:
Implement marking/unmarking of tasks and formatting tasks' status

4ndrelim committed:
Implement toggle methods of tasks

4ndrelim committed:
Complete marking and unmarking functionality

4ndrelim committed:
Refactor code for farmatting and collapsing print statements

4ndrelim committed:
make code consistent in extracting task number for mark/unmark

4ndrelim committed:
Implement event, todo, deadline subclasses and creation (sub) tasks

4ndrelim committed:
Update icon status whenever a task is added

4ndrelim committed:
add testing and expected output

4ndrelim committed:
Added exception for unknown commands and exception for missing todo description

4ndrelim committed:
Implement delete methosd for tasks

4ndrelim committed:
Add span class to format responses

4ndrelim committed:
Fix bug in the format of error msg and test

4ndrelim committed:
Add error handling for the other events

4ndrelim committed:
Refactor code to use enum and clean code

4ndrelim committed:
Add file mangement system

4ndrelim committed:
Complete save and load data features

4ndrelim committed:
Clean code and ignore saved output during testing

4ndrelim committed:
Refactor code to store dates as LocalDate objects

4ndrelim committed:
Implement feature to get all activities on a given date

4ndrelim committed:
Merge branch 'branch-level-7'

4ndrelim committed:
Resolve merge conflict and test code

4ndrelim committed:
Remove unnecessary exception class

4ndrelim committed:
Refactor code to include UI, Parser classes

4ndrelim committed:
Move classes into respective packages

4ndrelim committed:
Merge branch 'add-gradle-support'.
This is done to add gradle support to master

4ndrelim committed:
Add JUnit tests to test commands and tasks

4ndrelim committed:
Configure for JAR file

4ndrelim committed:
Add javadocs for all methods and improve variable naming

4ndrelim committed:
Align to coding standard

4ndrelim committed:
Implement find task feature

4ndrelim committed:
Merge branch 'branch-A-JavaDoc'

4ndrelim committed:
Merge branch-A-CodingStandard and resolve merge conflicts

4ndrelim committed:
Merge and resolve branch-Level-9 and enforce coding standards

YuanXun9834 commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,73 @@
+     * @param date Date to check against.
+     * @return boolean value that checks if the task falls on the specified date.
+     */
+    public boolean fallsOnDate(LocalDate date) {
```
Should the method name be changed to one that sounds more like a boolean? 👍

4ndrelim replied:
I think this already suggests a boolean? The method tells us whether a task indeed falls on a given date. Trying to intentionally add the conventional "is" / "can" / "has" sounds a little too forced and grammatically-off. Unless you have a better name for it in mind? haha

YuanXun9834 commented on src/main/java/duke/command/Parser.java:
```diff
@@ -0,0 +1,176 @@
+     * @return a boolean value that determines whether the program will continue or halt.
+     * @throws DukeException DukeException thrown if there is missing required information in the user's input.
+     */
+    public boolean execute(Command command, String echo) throws DukeException {
```
Should the various types of commands be abstracted out into different classes, so as to make use of dynamic binding?

4ndrelim replied:
Yes for extensibility. But i decided against it because 
1. takes time 
2. The initial parsing using regex differs for each command and i can't see much commonalities among the commands at least up till the current context.
Thanks!

YuanXun9834 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,59 @@
+ * @version CS2103T AY22/23 Semester 2
+ */
+public class Duke {
+    private FileManagement fileManager;
```
Great job in following the coding standard for your duke class ! 👍

4ndrelim replied:
Thanks!

YuanXun9834 commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,73 @@
+     * Displays formatted information about the task.
+     * @return String containing common information about a task.
+     */
+    public String getStatusIcon() {
```
I feel this method description is unnecessary as the class is already declared as an abstract class.

4ndrelim replied:
Ah okay noted on this point

YuanXun9834 reviewed (commented):
Overall great job in following the coding standard in your iP. I've provided a few minor feedback points and hope it helps :)

4ndrelim committed:
Set up checkstyle with Gradle

4ndrelim committed:
Further improve coding style

4ndrelim committed:
Adhere to coding style

4ndrelim committed:
Fix remaining coding standard violations

4ndrelim committed:
Merge branch 'branch-A-CheckStyle'

4ndrelim committed:
Add gui

4ndrelim committed:
Complete gui

4ndrelim committed:
Merge branch 'branch-Level-10'

4ndrelim committed:
Pass all style checks

4ndrelim committed:
Shift files

4ndrelim committed:
Configure build for gradle

4ndrelim committed:
Separate the handling for each command into individual classes

The user's input needs to be parsed and the program will subsequently
execute the methods corresponding to the command identified.

Combining the parsing and the execution of different commands in one
method makes it long and cluttered.

Hence, the code segment for each command is extracted and placed into
separate methods in their respective classes.

4ndrelim committed:
Use assert statements to catch bugs

4ndrelim committed:
Add assert statements to TaskList class

Some operations of TaskList class like toggleMark, delete require a
task number to execute.

Multiple assertions are added to this class to ensure the task number
supplied as argument is valid. Otherwise, it should have been handled
prior to the method call.

4ndrelim committed:
Abide by coding standard

4ndrelim committed:
Merge pull request #2 from 4ndrelim/branch-A-Assertions

Add assert statements

4ndrelim committed:
merge branch 'master' into 'branch-A-CodeQuality' and resolve conflicts

4ndrelim committed:
Merge pull request #3 from 4ndrelim/branch-A-CodeQuality

Adhere to coding standard

4ndrelim committed:
Fix bug when invalid date is provided

4ndrelim committed:
Implement get reminders feature

4ndrelim committed:
Merge branch 'branch-B-Reminders'

4ndrelim committed:
Improve gui and display

4ndrelim committed:
Add user guide

4ndrelim committed:
Resize chat window and update picture
